it/uk/gov/hmrc/incorporatedentityidentificationfrontend/controllers/CheckYourAnswersControllerISpec.scala

```diff
@@ -139,279 +139,329 @@ class CheckYourAnswersControllerISpec extends ComponentSpecHelper
+            journeyId = testJourneyId,
+            authInternalId = testInternalId,
+            continueUrl = testContinueUrl,
+            optServiceName = None,
+            deskProServiceId = testDeskProServiceId,
+            signOutUrl = testSignOutUrl,
+            businessEntity = LimitedCompany
+          ))
+
+          stubAuth(OK, successfulAuthResponse(Some(testInternalId)))
+          stubRetrieveCompanyProfileFromBE(testJourneyId)(status = OK, body = Json.toJsObject(testCompanyProfile))
+          stubRetrieveCtutr(testJourneyId)(status = OK, body = testCtutr)
+          stubStoreIdentifiersMatch(testJourneyId, identifiersMatch = false)(status = OK)
+          stubStoreBusinessVerificationStatus(testJourneyId, BusinessVerificationUnchallenged)(status = OK)
+          stubStoreRegistrationStatus(testJourneyId, RegistrationNotCalled)(status = OK)
+
+          stubValidateIncorporatedEntityDetails(
+            testCompanyNumber,
+            testCtutr
+          )(
+            status = BAD_REQUEST,
+            body = Json.obj(
+              "code" -> "NOT_FOUND",
+              "reason" -> "The back end has indicated that CT UTR cannot be returned"
+            )
+          )
+
+          lazy val result = post(s"$baseUrl/$testJourneyId/check-your-answers-business")()
+
+          result.status mustBe SEE_OTHER
+          result.header(LOCATION) mustBe Some(routes.JourneyRedirectController.redirectToContinueUrl(testJourneyId).url)
+          verifyStoreBusinessVerificationStatus(testJourneyId, BusinessVerificationUnchallenged)
+          verifyStoreRegistrationStatus(testJourneyId, RegistrationNotCalled)
+        }
+      }
+    }
+
+
+    "the Registered Society has provided ctutr and crn" should {
+      "return a redirect to the Business Verification Result page" in {
+        await(insertJourneyConfig(
+          journeyId = testJourneyId,
+          authInternalId = testInternalId,
+          continueUrl = testContinueUrl,
+          optServiceName = None,
+          deskProServiceId = testDeskProServiceId,
+          signOutUrl = testSignOutUrl,
+          businessEntity = RegisteredSociety
+        ))
+        stubAuth(OK, successfulAuthResponse(Some(testInternalId)))
+        stubRetrieveCompanyProfileFromBE(testJourneyId)(status = OK, body = Json.toJsObject(testCompanyProfile))
+        stubRetrieveCtutr(testJourneyId)(status = OK, body = testCtutr)
+        stubValidateIncorporatedEntityDetails(testCompanyNumber, testCtutr)(OK, Json.obj("matched" -> true))
+        stubStoreIdentifiersMatch(testJourneyId, identifiersMatch = true)(status = OK)
+        stubCreateBusinessVerificationJourney(testCtutr, testJourneyId)(status = CREATED)
+
+        lazy val result = post(s"$baseUrl/$testJourneyId/check-your-answers-business")()
+
+        result.status mustBe SEE_OTHER
+        result.header(LOCATION) mustBe Some(routes.BusinessVerificationController.startBusinessVerificationJourney(testJourneyId).url)
```

Comment:
Can we add verifies here to confirm the correct details have been stored please?